src/statsmodels/statsmodel.jl

```diff
@@ -48,8 +48,8 @@ for (modeltype, dfmodeltype) in ((:StatisticalModel, DataFrameStatisticalModel),
                                  (:RegressionModel, DataFrameRegressionModel))
     @eval begin
         function StatsBase.fit{T<:$modeltype}(::Type{T}, f::Formula, df::AbstractDataFrame,
-                                              args...; kwargs...)
-            mf = ModelFrame(f, df)
+                                              args...; contrasts::Dict = Dict(), kwargs...)
```

Comment:
Actually, now I'm confused:

```julia
julia> d = Dict(:x1p => EffectsCoding(),  :x2p => DummyCoding())
Dict{Symbol,Any} with 2 entries:
  :x2p => DataFrames.DummyCoding(Nullable{Any}(),Nullable{Array{T,1}}())
  :x1p => DataFrames.EffectsCoding(Nullable{Any}(),Nullable{Array{T,1}}())

julia> d::Dict{Symbol, AbstractContrasts}
ERROR: TypeError: typeassert: expected Dict{Symbol,DataFrames.AbstractContrasts}, got Dict{Symbol,Any}

```

So restricting the dict type with `contrasts::Dict{Symbol, AbstractContrasts}` throws an error in this (perfectly reasonable) case, even though you can convert fine:

```julia
julia> convert(Dict{Symbol, AbstractContrasts}, d)
Dict{Symbol,DataFrames.AbstractContrasts} with 2 entries:
  :x2p => DataFrames.DummyCoding(Nullable{Any}(),Nullable{Array{T,1}}())
  :x1p => DataFrames.EffectsCoding(Nullable{Any}(),Nullable{Array{T,1}}())
```